prov/rxm/src/rxm_init.c

```diff
@@ -38,25 +38,22 @@
 int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 		     struct fi_info *core_info)
 {
-	/* Support modes that ofi_rxm could handle */
-	if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
-		core_info->domain_attr->mr_mode |= FI_MR_LOCAL;
-	else
-		core_info->mode |= FI_LOCAL_MR;
+	core_info->mode |= FI_RX_CQ_DATA | FI_CONTEXT;
+	core_info->ep_attr->type = FI_EP_MSG;
 
-	/* The RxM FI_RMA implementation is pass-through but the provider
-	 * can handle FI_MR_PROV_KEY and FI_MR_VIRT_ADDR in its large message
-	 * transfer rendezvous protocol. */
-	if (!hints || !ofi_rma_target_allowed(hints->caps)) {
-		if (FI_VERSION_GE(version, FI_VERSION(1, 5)))
-			core_info->domain_attr->mr_mode |= OFI_MR_BASIC_MAP;
-		else
-			/* specify FI_MR_UNSPEC (instead of FI_MR_BASIC) so that
-			 * providers that support FI_MR_SCALABLE aren't dropped */
-			core_info->domain_attr->mr_mode = FI_MR_UNSPEC;
-	}
+	if (FI_VERSION_GE(version, FI_VERSION(1, 5))) {
+		core_info->domain_attr->mr_mode = OFI_MR_BASIC_MAP | FI_MR_LOCAL;
 
-	core_info->mode |= FI_RX_CQ_DATA | FI_CONTEXT;
+		if (hints && hints->domain_attr &&
+		    (hints->domain_attr->mr_mode != FI_MR_BASIC) &&
+		    (hints->domain_attr->mr_mode != FI_MR_SCALABLE)) {
+			core_info->domain_attr->mr_mode |=
+					hints->domain_attr->mr_mode;
+		}
+	} else {
+		core_info->domain_attr->mr_mode = FI_MR_UNSPEC;
+		core_info->mode |= FI_LOCAL_MR;
+	}
```

Comment:
mr_mode was being modified in 3 places in this call -- this consolidates those changes to one place.